### PR TITLE
feat(preview): Add download preview

### LINKED ISSUE #109: Add option to download final HTML
This could be an icon next to the copy to clipboard icon.

### PR DESCRIPTION
**Pull request by vcapretz**
Adds an icon next to "Copy" to download an HTML file with the preview code 

closes #109

### CODE REVIEW

File: packages/design-system/src/icons/icon-download.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import * as React from 'react';
+import { IconBase, IconElement, IconProps } from './icon-base';
+
+export const IconDownload = React.forwardRef<IconElement, Readonly<IconProps>>(
+  ({ ...props }, forwardedRef) => (
+    <IconBase ref={forwardedRef} {...props} width={18} height={18}>
+      <path
```
Comment (vcapretz) [inline packages/design-system/src/icons/icon-download.tsx:7]: this one is the one from react-feather, not sure we have another source of icons tho, let me know

File: packages/design-system/tsconfig.json [hunk 2]
```diff
@@ -1,5 +1,8 @@
 {
   "extends": "tsconfig/react-library.json",
+  "compilerOptions": {
+    "lib": ["dom", "es2017"]
```
Comment (vcapretz) [inline packages/design-system/tsconfig.json:4]: we need "dom" as a lib as well, the default one didn't include it and using browser APIs was not working

File: packages/design-system/src/icons/icon-download.tsx (hunk 1 above)
Comment (bukinoshita) [reply packages/design-system/src/icons/icon-download.tsx:7]: I'll put here the download icon

Comment (bukinoshita) [reply packages/design-system/src/icons/icon-download.tsx:7]:
```svg
<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
  <path d="M4.75 14.75V16.25C4.75 17.9069 6.09315 19.25 7.75 19.25H16.25C17.9069 19.25 19.25 17.9069 19.25 16.25V14.75" stroke="#A09FA6" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"/>
  <path d="M12 14.25V4.75" stroke="#A09FA6" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"/>
  <path d="M8.75 10.75L12 14.25L15.25 10.75" stroke="#A09FA6" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"/>
</svg>
```

File: packages/design-system/src/icons/icon-download.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import * as React from 'react';
+import { IconBase, IconElement, IconProps } from './icon-base';
+
+export const IconDownload = React.forwardRef<IconElement, Readonly<IconProps>>(
+  ({ ...props }, forwardedRef) => (
+    <IconBase ref={forwardedRef} {...props} width={18} height={18}>
```
Comment (bukinoshita) [inline packages/design-system/src/icons/icon-download.tsx:6]: Any reason the default width and height is 18?

**Review by bukinoshita — COMMENTED** [review]
Can you rename the PR?

```diff
- feature: Download HTML Preview
+ feat(preview): Add download preview
```

File: packages/design-system/src/icons/icon-download.tsx (hunk 1 above)
Comment (vcapretz) [reply packages/design-system/src/icons/icon-download.tsx:7]: updated!

File: packages/design-system/src/icons/icon-download.tsx (hunk 3 above)
Comment (vcapretz) [reply packages/design-system/src/icons/icon-download.tsx]: nah, it was just some value tweaking, now we have the correct svg we don't need these props anymore

File: packages/design-system/src/icons/icon-download.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import * as React from 'react';
+import { IconBase, IconElement, IconProps } from './icon-base';
+
+export const IconDownload = React.forwardRef<IconElement, Readonly<IconProps>>(
+  ({ ...props }, forwardedRef) => (
+    <IconBase ref={forwardedRef} {...props}>
+      <path
+        d="M4.75 14.75v1.5a3 3 0 0 0 3 3h8.5a3 3 0 0 0 3-3v-1.5M12 14.25v-9.5M8.75 10.75l3.25 3.5 3.25-3.5"
+        stroke="#A09FA6"
```
Comment (bukinoshita) [inline packages/design-system/src/icons/icon-download.tsx:9]:
```suggestion
        stroke="currentColor"
```

**Pull request merged**